tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs

```diff
@@ -123,6 +123,13 @@ internal static partial class ConfigurationKeys
         /// <seealso cref="TracerSettings.HeaderTags"/>
         public const string HeaderTags = "DD_TRACE_HEADER_TAGS";
 
+        /// <summary>
+        /// Configuration key for a map of metadata keys to tag names.
+        /// Automatically apply GRPC metadata values as tags on traces.
+        /// </summary>
+        /// <seealso cref="TracerSettings.HeaderTags"/>
+        public const string GrpcTags = "DD_TRACE_GRPC_TAGS";
```

Comment:
Did we consider reusing `DD_TRACE_HEADER_TAGS` for this? That name doesn't imply http-only.

Reply:
> Did we consider reusing DD_TRACE_HEADER_TAGS for this?

Yes, we (you, me + others 😄) discussed this. There wasn't a good consensus tbh. Most languages (java, ruby) always copy _all_ metadata, but have also said that's problematic so wouldn't suggest it. 

The trouble with reusing `DD_TRACE_HEADER_TAGS` IMO is that while that name doesn't _imply_ HTTP-only, it is, in fact only used for HTTP, and our documentation of `TracerSettings.HeaderTags` also says `"...which are applied to the root Span of incoming and outgoing HTTP requests"`. Other languages  also specifically tie `DD_TRACE_HEADER_TAGS` to being related to HTTP only, e.g. Java explicitly adds tags as `http.request.headers.<headername>` which strongly ties this setting to HTTP only (despite the naame) https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#:~:text=DD_TRACE_HEADER_TAGS

I'm not completely. averse to re-using the same setting for two things, although the overlap between GRPC and HTTP means reusing `DD_TRACE_HEADER_TAGS` might add an additional layer of confusion. It would also mean you can't record different data in GRPC calls than in HTTP calls. 

That said, it would certainly be easy enough to reuse `DD_TRACE_HEADER_TAGS` for now, and re-introduce `DD_TRACE_GRPC_TAGS` later if we so decide?

Reply:
No, yeah. In the end it might be better to rename `DD_TRACE_HEADER_TAGS`. Thanks.